getting-started/quickstart-guides/angular.md

```diff
@@ -0,0 +1,854 @@
+
+```jsx
+publicKey: "<YOUR_PUBLIC_KEY>",
+authenticationEndpoint: "<YOUR_AUTH_ENDPOINT e.g http:/localhost:3000/auth>"
+```
+
+### **Upload an image**
+
+For this, we will use the `ik-upload` component as well as a couple of event handlers for upload error and success, `onError` and `onSuccess` respectively. Let's use it in our `app.component.html` file:
+
+```js
+<ik-upload 
+    fileName="test.jpg" 
+    (onError)="handleUploadError($event)"
+    (onSuccess)="handleUploadSuccess($event)" 
+    >
+  </ik-upload>
+```
+This is how it looks in the UI:
+
+![Upload Image](<../../.gitbook/assets/angular/angular-sdk-file-upload-1.png>)
+
+**Direct file uploading from the browser**
+
+Let’s now upload an image by selecting a file from the file input.
+
+When you choose a file, the file is immediately uploaded. You can pass optional `onSuccess` and `onError` callback functions as props like we have.
+
+You can verify that file was successfully uploaded by checking the browser console. In case of success, it should print a success message, like this:
+
+![Upload Success Response](<../../.gitbook/assets/angular/angular-sdk-file-upload-2.png>)
+
+The response object would look similar to this (values may vary):
+
+```js
+{
+    "fileId": "63a2e985e809dd54b027a563",
+    "name": "test_RiCBw0ouh.jpg",
+    "size": 36919,
+    "versionInfo": {
+        "id": "63a2e985e809dd54b027a563",
+        "name": "Version 1"
+    },
+    "filePath": "/test_RiCBw0ouh.jpg",
+    "url": "https://ik.imagekit.io/yzyzyz/test_RiCBw0ouh.jpg",
+    "fileType": "image",
+    "height": 500,
+    "width": 500,
+    "thumbnailUrl": "https://ik.imagekit.io/yzyzyz/tr:n-ik_ml_thumbnail/test_RiCBw0ouh.jpg",
+    "AITags": null
+}
+```
+
+After a successful upload, you should see the newly uploaded image in the [Media Library](https://imagekit.io/dashboard#media-library) section of your ImageKit dashboard.
+
+If you don't see the image, check if there are any errors in the browser console log. Then verify whether the API [private key](../../api-reference/api-introduction/api-keys.md#private-key) has been configured correctly in the server app and if the server app is running.
+
+**Fetching uploaded file**
+
+Fetch uploaded image and show in UI using `ik-image` with the `filePath` returned in the upload response.
+
+```jsx
+<ik-image path="/test_RiCBw0ouh.png" ></ik-image>
+```
+
+The app should display your uploaded image correctly!
+
+## **Advanced file upload**
+
+A more detailed example of how to use the file upload component (and an explanation of each advanced feature) is presented below:
+
+{% code title="src/app/app.component.ts" %}
+```js
+import { Component } from '@angular/core';
+import { Transformation } from 'imagekit-javascript/dist/src/interfaces/Transformation';
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.css']
+})
+export class AppComponent {
+    validateFileFunction(res: any) {
```

Comment:
Code snippets should be properly formatted thought this PR.

Reply:
They are formatted - i.e spacing, indents are copied pasted from VSC.

Reply:
Updated